code/modules/projectiles/ammo_datums.dm

```diff
@@ -1670,13 +1670,21 @@ datum/ammo/bullet/revolver/tp44
 /datum/ammo/rocket/do_at_max_range(turf/T, obj/projectile/P)
 	drop_nade(T.density ? P.loc : T)
 
+/datum/ammo/rocket/dumbfire
+	damage = 50
+	flags_ammo_behavior = AMMO_ROCKET|AMMO_SUNDERING // We want this one to specifically go over onscreen range.
+
+/datum/ammo/rocket/drop_nade(turf/T)
+	explosion(T, 0, 7, 0, 2)
```

Comment:
You are mixing datums order here. do all parent rocket stuff then define children